classes/Hook.php

```diff
@@ -315,34 +314,67 @@ private static function getHookAliasesFor($hookName)
         return Cache::retrieve($cacheId);
     }
 
+    /**
+     * Returns a list of hook names, indexed by alias.
+     *
+     * @return array Array of hook names, indexed by lower case alias
+     *
+     * @throws PrestaShopDatabaseException
+     */
+    private static function getHookAliasDictionary()
+    {
+        $cacheId = 'hook_alias';
+
+        if (!Cache::isStored($cacheId)) {
+            $databaseResults = Db::getInstance()->executeS('SELECT * FROM `' . _DB_PREFIX_ . 'hook_alias`');
+            $hooksByAlias = array();
+            if ($databaseResults) {
+                foreach ($databaseResults as $record) {
+                    $hooksByAlias[strtolower($record['alias'])] = $record['name'];
+                }
+            }
+            Cache::store($cacheId, $hooksByAlias);
+
+            return $hooksByAlias;
+        }
+
+        return Cache::retrieve($cacheId);
+    }
+
     /**
      * Check if a hook or one of its old names is callable on a module.
      *
      * @since 1.7.1.0
      *
-     * @param $module
-     * @param $hookName
+     * @param static $module
+     * @param string $hookName
      *
      * @return bool
      */
     private static function isHookCallableOn($module, $hookName)
     {
-        $aliases = Hook::getHookAliasesFor($hookName);
-        $aliases[] = $hookName;
+        $aliases = array_merge(
+            [$hookName],
```

Comment:
@eternoendless i'm still lost whether we can or not use shorthand arrays :D

Reply:
We can!

Reply:
And we love it!